lio_sam:

  # voxel filter paprams
  odometrySurfLeafSize: 0.2   #0.2                          # default: 0.4 - outdoor, 0.2 - indoor  
  mappingCornerLeafSize: 0.1  #0.1                   # default: 0.2 - outdoor, 0.1 - indoor 
  mappingSurfLeafSize: 0.2    #0.2                   # default: 0.4 - outdoor, 0.2 - indoor   
  mappingFullLeafSize: 0.05    #0.01                  # default: 0.01  save map

  # robot motion constraint (in case you are using a 2D robot)
  z_tollerance: 50                              # meters  1000
  rotation_tollerance: 100                      # radians  1000

  # CPU Params
  numberOfCores: 8                              # number of cores for mapping optimization
  mappingProcessInterval: 0.1  #0.2             # seconds, regulate mapping frequency ,default: 0.15

  # Surrounding map
  surroundingkeyframeAddingDistThreshold: 0.5         # meters, regulate keyframe adding threshold
  surroundingkeyframeAddingAngleThreshold: 0.2        # radians, regulate keyframe adding threshold  0.2
  surroundingKeyframeDensity: 1.0        # meters, downsample surrounding keyframe poses  2.0   0.5
  surroundingKeyframeSearchRadius: 50.0               # meters, within n meters scan-to-map optimization (when loop closure disabled); default: 50

  # Loop closure
  loopClosureEnableFlag: true
  loopClosureFrequency: 1.0                     # Hz, regulate loop closure constraint add frequency default: 1.0
  surroundingKeyframeSize: 20.0                 # submap size (when loop closure enabled) default: 50.0   20.0
  historyKeyframeSearchRadius: 5.0 #35.0       # meters, key frame that is within n meters from current pose will be considerd for loop closure default: 15.0
  historyKeyframeSearchTimeDiff: 60.0 #40.0     # seconds, key frame that is n seconds older will be considered for loop closure default: 30
  historyKeyframeSearchNum: 30                  # number of hostory key frames will be fused into a submap for loop closure default: 25
  historyKeyframeFitnessScore: 0.15              # icp threshold, the smaller the better alignment 0.3

  # Visualization
  globalMapVisualizationSearchRadius: 100.0     # meters, global map visualization radius
  globalMapVisualizationPoseDensity: 10.0       # meters, global map visualization keyframe density
  globalMapVisualizationLeafSize: 0.1 #1.0      # meters, global map visualization cloud density

  # visual iktree_map   
  visulize_IkdtreeMap: true

  # reconstruct iktree_map  
  recontructKdTree: false
  angleThreshold: 0.1 # angleThreshold to judge reconstruct or not
  transThreshold: 0.1 # transThreshold to judge reconstruct or not
  # back-end
  backEndEnable: true
  # update lio state or not 
  updateLio: false

  # Export settings
  saveFrameInterval: 15
  savePCD: true                        # https://github.com/TixiaoShan/LIO-SAM/issues/3
  savePCDDirectory: "/map_save/fastlio_map/"       # in your home folder, starts and ends with "/". Warning: the code deletes "LOAM" folder then recreates it. See "mapOptimization" for implementation

gps:
  bUseGps: true # use gps or not 
  debugGps: true # debug
  addGps: true # add gps factor
  gpsInitOrientation: false # just use gps to init orientation
  useGpsElevation: true #use gps Elevation or not 
  useImuHeadingInitialization: true #use gps yaw or not, no use
  updateLioOnce: true # update lio once or not
  rpyInitMode: 2 #0:imu, 1:beidou, 2:set use to init yaw
  gpsTopic: "/gps_topic" #"/beidou" #"/gnss" #
  rpyTopic: "/bd_rpy"
  imuTopic: "/imu_topic"
  gpsAltitudeNoiseScore: 10.0
  gpsLatitudeNoiseScore: 100.0
  gpsFrequence: 10 # Frequence of gps
  gpsInitPointCnt: 20 # number of gps, first add
  gpsCovThreshold: 8.0 # covariance of gps to add
  gpsDistance: 0.5 # two gps distance to add a gps factor
  RPYGPS2Imu: [0.0, 0.0, 0.0] # rotation from gps to imu